mesonbuild/options.py

```diff
@@ -889,7 +890,19 @@ def get_value_object_and_value_for(self, key: OptionKey) -> T.Tuple[AnyOptionTyp
                 computed_value = vobject.validate_value(self.augments[key])
         return (vobject, computed_value)
 
-    def get_value_for(self, name: 'T.Union[OptionKey, str]', subproject: T.Optional[str] = None) -> ElementaryOptionValues:
+    def get_value_for(self, key: OptionKey, type_: T.Type[ElementaryOptionTypes],
+                      *, fallback: T.Optional[ElementaryOptionTypes] = None) -> ElementaryOptionTypes:
```

Comment:
Maybe `default` would be more in line to how the same concept is named in the Pyrthon APIs than `fallback`